onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,177 @@
+                # Retrieve or create user's profile.
+                (
+                    user_profile,
+                    created,
+                ) = UserProfile.objects.get_or_create(user_id=xform['user_id'])
+
+                # Some old profiles don't have metadata
+                if user_profile.metadata is None:
+                    user_profile.metadata = {}
+
+                # Set the flag to true if it was never set.
+                if not user_profile.metadata.get('submissions_suspended'):
+                    # We are using the flag `submissions_suspended` to prevent
+                    # new submissions from coming in while the
+                    # `attachment_storage_bytes` is being calculated.
+                    user_profile.metadata['submissions_suspended'] = True
+                    user_profile.save(update_fields=['metadata'])
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            # We cannot get the sum of all attachments for on xform because
+            # we need to make two joins with Instance and XForm models.
+            # It is really slow on big databases. So, to avoid that, we make
+            # another extra query to get all the instance ids to pass it to
+            # Attachment queryset.
+            instance_ids = Instance.objects.values_list('pk', flat=True).filter(
+                xform_id=xform['pk']
+            )
+
+            # It does not seem to have a real limit of number of parameters in
+            # "IN" clause in PostgreSQL but for safety, we use chunks to pass
+            # to `instance_id__in` just in case a form has a huge numbers
+            # of submissions.
+            # See https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause
+            instance_ids_count = len(instance_ids)
+            max_ids_per_query = 5000
+            chunks = [
+                instance_ids[x:x + max_ids_per_query]
+                for x in range(0, instance_ids_count, max_ids_per_query)
+            ]
+            form_attachments_total = 0
+            for idx, chunk in enumerate(chunks):
+                if self.verbosity > 1:
+                    self.stdout.write(
+                        f'\tCalculating total: {idx+1}/{len(chunks)} instance ID'
+                        f' chunks'
```

Comment:
We have enough info to show a percentage if we keep this approach, which might be more helpful since someone wouldn't know the chunk size without reading the code

Reply:
Not accurate anymore. This code is gone